app/assets/javascripts/directives/file-upload-modal.coffee

```diff
@@ -0,0 +1,56 @@
+angular.module('infish').directive 'fileUploadModal', ($timeout) ->
+  templateUrl: 'file-upload-modal.html'
+  controller: ($scope) ->
+    modalOpen = false
+
+    $scope.ifShowModal = -> modalOpen
+    $scope.showModal = -> modalOpen = true
+    $scope.closeModal = -> modalOpen = false
+    $scope.toggleModal = -> modalOpen = !modalOpen
+
+
+    addToForm = (question, answers) ->
+      states = []
+      states.push (question[i] == '1') for i in [1..4]
+      question = question[5]
+
+      $scope.addNewFilledQuestion(question, answers, states)
+      $scope.$apply()
+
+    parseFile = (content) ->
+      regex = /// ^ #begin of line
+        X(\d)(\d)(\d)(\d)[\r\n]{1,2}
+        [\d.\s]*(.*):*[\r\n\t\s]+
+         ///i      
+
+      regex2 = /[a-j]\.[\s]*(.*)[\r\n\t\s]*/g
```

Comment:
Please use more expressive variables names, `regex` and `regex2` don't mean anything.